classtest.js

```diff
@@ -1,87 +1,152 @@
+var colTotals = [];
+  //---starts the total at zero so that we can calculate totals within the for loops. resetting to zero allows us to differentiate different rows---//
 var total = 0;
-//----------------------For Loop to run through the times------------
-
-var rowOneEl = document.createElement('tr');
-var anything = document.createElement('th');
-anything.textContent = timeSlot.label;
-rowOneEl.appendChild(anything);
-
-for (var k = 0; k < timeSlot.times.length ; k++){
-  var timeLabelEl = document.createElement('th');
-  var currentTime = timeSlot.times;
-  timeLabelEl.textContent = currentTime[k];
-  rowOneEl.appendChild(timeLabelEl);
-};
-var totalHead = document.createElement('th');
-totalHead.textContent = timeSlot.totalLabel;
-rowOneEl.appendChild(totalHead);
-
-tableEl.appendChild(rowOneEl);
-
+  //---Creates the table element in the html that we insert into-----//
+var tableEl = document.createElement('table');
 document.body.appendChild(tableEl);
 
-for(var i = 0; i < stores.length; i++){
-  var currentStore = stores[i];
+//------------------------FUNCTIONS----------------------------//
+//---------------function to calculate total cookies per hour for each store------
 
-  var rowEl = document.createElement('tr');
+//------------???? what is this again ?????-------------//
 
+// -----------------function to create new cookie store objects.(use of the .this method here)-------//
+function CookieStore(name, minCustomers, maxCustomers, avgCookies) {
+  this.name = name;
+  this.minCustomers = minCustomers;
+  this.maxCustomers = maxCustomers;
+  this.avgCookies = avgCookies;
+  this.hourlySales = [];
+}
+CookieStore.prototype.getAvgCookieCount = function (){
+  var range = this.maxCustomers - this.minCustomers;
+  return Math.floor(Math.random() * range + this.minCustomers);
+};
+CookieStore.prototype.renderRow = function() {
+  var rowEl = document.createElement('tr');
   var nameEl = document.createElement('th');
-  nameEl.textContent = currentStore.name;
+  nameEl.textContent = this.name + ':';
   rowEl.appendChild(nameEl);
 
-
+//--for loop to produce cph stats for each time slot for this row (it is inside the store loop so it will go 16 times five different times....)
   for(var j = 0; j < timeSlot.times.length ; j++){
+    var storeRandomCPH = this.getAvgCookieCount();
     var timeEl = document.createElement('td');
-    timeEl.textContent = cph();
+    timeEl.textContent = storeRandomCPH;
     rowEl.appendChild(timeEl);
-    total += cph();
+    total += storeRandomCPH;
+    this.hourlySales.push(storeRandomCPH);
+    console.log(this.hourlySales);
   }
-  //log to dom the total in a column
-  console.log(currentStore.name + ' ' + total);
+  //---render to dom the total in a column---//
   tableEl.appendChild(rowEl);
   var totalEl = document.createElement('td');
   totalEl.textContent = total;
   rowEl.appendChild(totalEl);
   total = 0;
 };
+var colTotalAdded = document.createElement('tr');
```

Comment:
Try to isolate little chunks of code like this into functions, then call those functions, instead of having functionality floating out at the root level.